### PR TITLE
MSDKUI-1634: Use current Locale to set Speed View and Speed Limit Vie…

### PR DESCRIPTION
**Pull request by tstaszak89**
…w unit

Added support of unit systems in lib and used it in app.

Signed-off-by: Tadeusz Staszak <36914750+tstaszak89@users.noreply.github.com>

### CODE REVIEW

File: MSDKUIDemo/MSDKUIApp/src/main/java/com/here/msdkuiapp/common/UnitSystemFromLocaleUtil.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ */
+object UnitSystemFromLocaleUtil {
+
+    private const val UNITED_STATES_COUNTRY_CODE = "US"
+    private const val UNITED_KINGDOM_COUNTRY_CODE = "GB"
+
+    /**
+     * Returns unit system for current locale.
+     *
+     * @return unit system [UnitSystems] for current locale.
```
Comment (datasun) [inline MSDKUIDemo/MSDKUIApp/src/main/java/com/here/msdkuiapp/common/UnitSystemFromLocaleUtil.kt]: Plural.

File: MSDKUIDemo/MSDKUIApp/src/main/java/com/here/msdkuiapp/common/UnitSystemFromLocaleUtil.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * Util class to get unit system for current locale.
+ */
+object UnitSystemFromLocaleUtil {
+
+    private const val UNITED_STATES_COUNTRY_CODE = "US"
+    private const val UNITED_KINGDOM_COUNTRY_CODE = "GB"
+
+    /**
+     * Returns unit system for current locale.
```
Comment (datasun) [inline MSDKUIDemo/MSDKUIApp/src/main/java/com/here/msdkuiapp/common/UnitSystemFromLocaleUtil.kt]: Plural.

File: MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/common/VelocityFormatterUtil.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import com.here.msdkui.common.measurements.UnitSystems;
+
+/**
+ * A convenience class to convert velocity in meter per second to velocity in a different unit.
+ */
+public final class VelocityFormatterUtil {
+    private VelocityFormatterUtil() {}
+
+    /**
+     * Converts velocity expressed in meters per second to specified unit system.
```
Comment (datasun) [inline MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/common/VelocityFormatterUtil.java]: Plural?

File: MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/common/VelocityFormatterUtil.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+import android.content.Context;
+
+import com.here.msdkui.R;
+import com.here.msdkui.common.measurements.Measurement;
+import com.here.msdkui.common.measurements.MeasurementUnit;
+import com.here.msdkui.common.measurements.SpeedConverter;
+import com.here.msdkui.common.measurements.UnitSystems;
+
+/**
+ * A convenience class to convert velocity in meter per second to velocity in a different unit.
```
Comment (datasun) [inline MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/common/VelocityFormatterUtil.java]: meters

File: MSDKUIKit/MSDKUILib/src/main/res/layout/guidance_current_speed_panel.xml [hunk 5]
```diff
@@ -27,5 +27,5 @@
         android:id="@+id/guidance_current_speed_unit"
         style="?attr/guidanceCurrentSpeedPanelSecondaryText"
         android:layout_below="@+id/guidance_current_speed_value"
-        android:text="@string/msdkui_unit_km_per_h" />
+        android:text="@null" />
```
Comment (imvjk) [inline MSDKUIKit/MSDKUILib/src/main/res/layout/guidance_current_speed_panel.xml]: you can just remove this line.. no?

File: MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/common/VelocityFormatterUtil.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+ */
+public final class VelocityFormatterUtil {
```
Comment (imvjk) [inline MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/common/VelocityFormatterUtil.java]: may be we can also think some good name?

File: MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/common/VelocityFormatterUtil.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+     * Returns abbreviation of velocity string in specified unit system.
+     *
+     * @param context
+     *         the required context.
+     * @param system
+     *         the unit system {@link UnitSystems}.
+     *
+     * @return abbreviation of velocity string in specified unit system.
+     */
+    public static String getVelocityString(final Context context, final UnitSystems system) {
```
Comment (imvjk) [inline MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/common/VelocityFormatterUtil.java]: here too abt name

File: MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/guidance/GuidanceEstimatedArrivalView.java [hunk 8]
```diff
@@ -40,6 +41,7 @@
 public class GuidanceEstimatedArrivalView extends FrameLayout {
```
Comment (imvjk) [inline MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/guidance/GuidanceEstimatedArrivalView.java]: in general, adding setter getter for unit system in everyview i would say to make a base view which allow this setter getter.. so u just have to extend this view from all other view which need unit system. i also see some view are extending framelayout and some relative layout. although we can make it consistence too

File: MSDKUIDemo/MSDKUIApp/src/main/java/com/here/msdkuiapp/common/UnitSystemFromLocaleUtil.kt [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+package com.here.msdkuiapp.common
+
+import com.here.msdkui.common.measurements.UnitSystems
+import java.util.Locale
+
+/**
+ * Util class to get unit system for current locale.
+ */
+object UnitSystemFromLocaleUtil {
```
[the remark this answers is not in the document]
Comment (datasun) [reply MSDKUIDemo/MSDKUIApp/src/main/java/com/here/msdkuiapp/common/UnitSystemFromLocaleUtil.kt]: UnitSystem.getUnitForLocale

File: MSDKUIDemo/MSDKUIApp/src/main/java/com/here/msdkuiapp/common/UnitSystemFromLocaleUtil.kt [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+    private const val UNITED_STATES_COUNTRY_CODE = "US"
+    private const val UNITED_KINGDOM_COUNTRY_CODE = "GB"
+
+    /**
+     * Returns unit system for current locale.
+     *
+     * @return unit system [UnitSystems] for current locale.
+     */
+    fun get(): UnitSystems {
```
[the remark this answers is not in the document]
Comment (datasun) [reply MSDKUIDemo/MSDKUIApp/src/main/java/com/here/msdkuiapp/common/UnitSystemFromLocaleUtil.kt]: getUnitForLocale(locale)

File: MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/common/BaseView.java [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+     */
+    public BaseView(Context context, @Nullable AttributeSet attrs) {
+        this(context, attrs, 0);
+    }
+
+    /**
+     * Constructs a new instance.
+     *
+     * @param context
+     *         the required {@link Context}.
```
Comment (datasun) [inline MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/common/BaseView.java]: Super ctor doc can be left out like elsewhere in the repo, but minor ...

File: MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/routing/ManeuverList.java [hunk 12]
```diff
@@ -76,6 +78,28 @@ public ManeuverList(final Context context, final AttributeSet attrs, final int d
         init();
     }
 
+    /**
+     * Sets unit system of this list items.
```
Comment (datasun) [inline MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/routing/ManeuverList.java]: for the items of this list

File: MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/routing/ManeuverList.java [hunk 13]
```diff
@@ -76,6 +78,28 @@ public ManeuverList(final Context context, final AttributeSet attrs, final int d
...
+     */
+    public void setUnitSystem(UnitSystem unitSystem) {
+        mUnitSystem = unitSystem;
+        if (mAdapter != null) {
+            mAdapter.setUnitSystem(mUnitSystem);
+        }
+    }
+
+    /**
+     * Returns current unit system of this list items.
```
Comment (datasun) [inline MSDKUIKit/MSDKUILib/src/main/java/com/here/msdkui/routing/ManeuverList.java]: same

**Pull request merged**